sandbox/lua/modules/line_protocol.lua

```diff
@@ -0,0 +1,246 @@
+    Configuration is implemented in the encoders that utilize this module.
+    This function takes no arguments.
+
+    *Arguments*
+        - config (table or nil)
+            Table of config option overrides that come from the client
+            of this API.  Defaults for this module are defined within
+            the set_config function and clients implementing this API
+            can reference it for available options.
+
+    *Return*
+        A table of Carbon line protocol messages ready to be sent to
+        a Carbon server after being looped through in an encoder implementing
+        this API.
+
+**influxdb_line_msg(config)**
+    Wrapper function that calls others within this module and the field_util
+    module to generate a table of InfluxDB line protocol messages that are
+    derived from the base or dynamic fields in a Heka message. Base fields or
+    dynamic fields can be used in the metric name portion of the message and
+    included as tags if desired.  Configuration is implemented in the encoders
+    that utilize this module.  This function takes no arguments.
+
+    *Arguments*
+        - config (table or nil)
+            Table of config option overrides that come from the client
+            of this API.  Defaults for this module are defined within
+            the set_config function and clients implementing this API
+            can reference it for available options.
+
+    *Return*
+        A table of InfluxDB line protocol messages ready to be sent to
+        an InfluxDB server after being looped through in an encoder
+        implementing this API.
+
+--]=]
+
+local interp = require "msg_interpolate"
+local field_util = require "field_util"
+local math = require "math"
+local read_config = read_config
+local read_message = read_message
+local read_next_field = read_next_field
+local pairs = pairs
+local string = require "string"
+local table = require "table"
+local tostring = tostring
+local type = type
+
+local M = {}
+setfenv(1, M) -- Remove external access to contain everything in the module.
+
+local function influxdb_kv_fmt(string)
+    return tostring(string):gsub("([ ,])", "\\%1")
+end
+
+local function points_tags_tables(config)
+    local name_prefix = ""
+    if config["name_prefix"] then
+        name_prefix = interp.interpolate_from_msg(config["name_prefix"])
+    end
+    local name_prefix_delimiter = config["name_prefix_delimiter"] or ""
+    local used_tag_fields = config["used_tag_fields"]
+    local skip_fields = config["skip_fields"]
+
+    -- Initialize the tags table, including base field tag values in
+    -- list if the magic **all** or **all_base** config values are
+    -- defined.
+    local tags = {}
+    if not config["carbon_format"]
+        and (config["tag_fields_all"]
+        or config["tag_fields_all_base"]
+        or config["used_tag_fields"]) then
+            for field in pairs(field_util.base_fields_tag_list) do
```

Comment:
Removed the separate function for this and moved it back into this function, simplifying the implementation based on what it actually needed to achieve (looping through "taggable" base fields, as defined by the base_fields_tag_list table.